splice_spark2/src/main/scala/com/splicemachine/spark2/splicemachine/ConnectionManager.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2012 - 2021 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.splicemachine.spark2.splicemachine
+
+import java.sql.{Connection, Savepoint}
+
+import org.apache.spark.sql.execution.datasources.jdbc.{JDBCOptions, JdbcUtils}
+
+@SerialVersionUID(20210317241L)
+class ConnectionManager(url: String) extends Serializable {
+  
+  private var con: Option[Connection] = None
+  
+  def autoCommitting(): Boolean = con.isEmpty
+  def transactional(): Boolean = ! autoCommitting
+
+  def getConnection(): Connection = if( autoCommitting ) {
+    createConnection()
+  } else {
```

Comment:
assign newly created connection to conn?

Reply:
Changed to use one connection instead of creating new connections in commit https://github.com/splicemachine/spliceengine/pull/5322/commits/158fc092a52db2371ed3b3364576299cd78df634